General attribute docs improvements

Havvy opened the pull request:
Each commit does its own small thing.

This builds on top of #350, so merge that before taking a look at this.

matthewjasper commented on src/items/implementations.md:
```diff
@@ -136,5 +136,11 @@ impl Seq<bool> for u32 {
+## Attributes in Implementations
+
+Implementations may contain inner [attributes] inside the brackets that contain
+the associated items. They must come before the associated items.
```
Could there be a sentence on what attributes are allowed on impls?

Havvy commented:
Addressed comments. Also added which attributes are allowed on block expressions. And apparently, as per the example I included on the block expressions page, block expressions also allow outer attributes and the only place you'd be able to tell that isn't overlapping with statements taking attributes is the return expression of a block expression. Go figure.

I'll add which attributes are allowed on functions / modules / the main module/ other items / variants and fields later.

Havvy commented:
Rebased against master due to merge conflict.

Havvy commented:
Rebased again for another merge conflict.
